src/Npgsql/NpgsqlCommand.cs

```diff
@@ -194,7 +197,10 @@ public new NpgsqlConnection Connection
                 if (_transaction != null && _connection != null && _connection.Connector != null && _connection.Connector.InTransaction)
                     throw new InvalidOperationException("The Connection property can't be changed with an uncommited transaction.");
 
-                _connection = value;
+                _connection = State == CommandState.Idle
+                    ? value
+                    : throw new InvalidOperationException("An open data reader exists for this command.");
```

Comment:
I prefer `if` in such cases, but it's up to you.

Reply:
I slightly prefer throwing from conditional expressions, but not by much. Basically almost any conditional expression is preferable to heavy if statements IMHO :) Until it gets too complicated/unreadable.

Leaving as-is.